rotkehlchen/chain/ethereum/modules/uniswap/utils.py

```diff
@@ -253,3 +273,319 @@ def historical_uniswap_v2_lp_price(
+                    )
+                    for index in chunk
+                ],
+            )
+        except RemoteError as e:
+            log.error(UNISWAP_V3_ERROR_MSG.format('nft contract token ids', str(e)))
+            continue
+
+        tokens_ids = [
+            nft_manager_contract.decode(   # pylint: disable=unsubscriptable-object
+                result=data[1],
+                method_name='tokenOfOwnerByIndex',
+                arguments=[address, index],
+            )[0]
+            for index, data in enumerate(tokens_ids_multicall) if data[0] is True
+        ]
+        try:
+            positions_multicall = multicall_2(
+                ethereum=ethereum,
+                require_success=False,
+                calls=[
+                    (
+                        UNISWAP_NFT_MANAGER_ADDRESS,
+                        nft_manager_contract.encode('positions', [token_id]),
+                    )
+                    for token_id in tokens_ids
+                ],
+            )
+        except RemoteError as e:
+            log.error(UNISWAP_V3_ERROR_MSG.format('nft contract positions', str(e)))
+            continue
+        positions = [
+            nft_manager_contract.decode(
+                result=data[1],
+                method_name='positions',
+                arguments=[tokens_ids[index]],
+            )
+            for index, data in enumerate(positions_multicall) if data[0] is True
+        ]
+        pool_addresses = [
+            compute_pool_address(
+                factory_address=UNISWAP_V3_FACTORY_ADDRESS,
+                token0_address=position[2],
+                token1_address=position[3],
+                fee=position[4],
+            )
+            for position in positions
+        ]
+        pool_contracts = [
+            EthereumContract(
+                address=pool_address,
+                abi=UNISWAP_V3_POOL_ABI,
+                deployed_block=UNISWAP_V3_DEPLOYED_BLOCK,
+            )
+            for pool_address in pool_addresses
+        ]
+        try:
+            slots_0_multicall = multicall_2(
+                ethereum=ethereum,
+                require_success=False,
+                calls=[
+                    (entry[0], entry[1].encode('slot0'))
+                    for entry in zip(pool_addresses, pool_contracts)
+                ],
+            )
+        except RemoteError as e:
+            log.error(UNISWAP_V3_ERROR_MSG.format('pool contract slot0', str(e)))
+            continue
+        slots_0 = [
+            entry[0].decode(entry[1][1], 'slot0')
+            for entry in zip(pool_contracts, slots_0_multicall) if entry[1][0] is True
+        ]
+        tokens_a = [
+            ethereum.get_basic_contract_info(to_checksum_address(position[2]))
+            for position in positions
+        ]
+        tokens_b = [
+            ethereum.get_basic_contract_info(to_checksum_address(position[3]))
+            for position in positions
+        ]
```

Comment:
In this function you do **a lot** of list comprehensions. Why? The function becomes bigger, and the code is slower. Take here for example. You iterate `positions` twice in 8 lines.

Look at my suggestion. It accomplishes the same thing and iterates once in 4 lines.
```suggestion
        tokens_a = tokens_b = []
        for position in positions:
            tokens_a.append(ethereum.get_basic_contract_info(to_checksum_address(position[2])))
            tokens_b.append(ethereum.get_basic_contract_info(to_checksum_address(position[3])))
```

Reply:
Oh something that was pointed out to me! I made a mistake here.

Should be `tokens_a, tokens_b = [], []`

Otherwise tokens_a and tokens_b point to the same list.